internal/abci/interfaces.go

```diff
@@ -50,8 +51,8 @@ type TxApp interface {
 	ChainInfo(ctx context.Context) (int64, []byte, error)
 	ApplyMempool(ctx context.Context, tx *transactions.Transaction) error
 	// Begin signals that a new block has begun.
-	Begin(ctx context.Context) error
-	Finalize(ctx context.Context, blockHeight int64) (appHash []byte, validatorUpgrades []*types.Validator, err error)
+	Begin(ctx context.Context) (sql.OuterTx, error)
```

Comment:
*If* we want to pass the transaction up to abci and on to the broadcaster, we'd give it the minimum required interface.  `sql.OuterTx` gives too much power.

`voting.OutstandingResolutions` only needs a `sql.Executor`.  The `EventBroadcaster` says it only needs a `sql.DB`, which is probably good for more generality (allow Execute or making a transaction/savepoint), so I think we should go with `sql.DB` for this `Begin` interface.

Still thinking through how the evt store's broadcast fn now needs voting store.  Separation of concerns between the two stores is getting hard to track.

Reply:
Event store was depending already on VoteStore to ensure that it doesn't add processed events into the Eventstore. I took it one more step ahead with 2 other APIs `GetObservedEvents` and `GetEvents`